src/NServiceBus.Core/Sagas/ActiveSagaInstance.cs

```diff
@@ -170,7 +165,18 @@ static object GetDefault(Type type)
             return null;
         }
 
-        Dictionary<string, object> correlatedPropertyValues = new Dictionary<string, object>();
+        internal IDictionary<string, object> CorrelationProperties
+        {
+            get { return correlationProperties.ToDictionary(cp => cp.PropertyInfo.Name, cp => cp.Value); }
```

Comment:
I'm about to change the api to only pass a single prop in a different pull so given that this is only called once I propose we leave it as is for now?